src/software/simulated_tests/ball_enters_region.py

```diff
@@ -0,0 +1,57 @@
+import pytest
+
+import software.geom.geometry as geom
+from proto.geometry_pb2 import Angle, AngularVelocity, Point, Vector
+from proto.tbots_software_msgs_pb2 import Vision
+from proto.validation_pb2 import ValidationGeometry, ValidationProto, ValidationStatus
+from proto.vision_pb2 import BallState, RobotState
+from proto.world_pb2 import SimulatorTick, World, WorldState
+
+from software.simulated_tests.validation import (
+    Validation,
+    create_validation_geometry,
+    create_validation_types,
+)
+
+
+class BallEntersRegion(Validation):
+
+    """Checks if a ball enters any of the provided regions."""
+
+    def __init__(self, regions=[]):
```

Comment:
Setting default parameter to an empty list can be dangerous since every instance of `BallEntersRegion` will refer to the same list.
It's preferred to do initialization as so:
```python
def __init__(self, regions=None):
  self.regions = regions if regions != None else []
```
See this [post](https://stackoverflow.com/questions/19497879/using-a-empty-list-as-default-parameter-in-init-why-the-list-in-all-instanc) for more details.

Reply:
Damn, sucks that this is even a problem, TIL thank you for pointing it out

fixed!